### PR TITLE
Change anime synopsis xpath

### LINKED ISSUE #495: Anime synopsis missing newlines between paragraphs
**Problem**

Anime synopsis are missing newlines between paragraph due to usage of meta description for XPath.

**Reproduction**

Example [MAL ID 6](https://api.jikan.moe/v4/anime/6)

**Solution**

1. Update XPath for Synopsis to no longer use meta tag for description, but actual content: `//p[@itemprop='description']`.
2. Update existing [test](https://github.com/jikan-me/jikan/blob/ae622d2e9954e91872866b492c4c4fefa7581a09/test/JikanTest/Parser/Anime/AnimeParserTest.php#L108) by changing the `assertStringContainsString` to `assertEquals` on exact description including newlines.

### PR DESCRIPTION
**Pull request by ToshY**
Fixes #495

### CODE REVIEW

**Comment by ToshY** [comment]
Unrelated tests failing

**Comment by pushrbx** [comment]
We need to test whether the JSON serializer in Jikan API drops these newlines or replaces them with `\n`.

**Review by pushrbx — CHANGES_REQUESTED** [review]
Please add the same change for the other media types too.

**Comment by ToshY** [comment]
Hey @pushrbx 

> Please add the same change for the other media types too.

Can you clarify which other media types? The `MangaParser` and `AnimeEpisodeParser` are also using the same XPath with meta tag, so I can update those too if that's what you're referring to.

**Comment by pushrbx** [comment]
> Hey @pushrbx
> 
> > Please add the same change for the other media types too.
> 
> Can you clarify which other media types? The `MangaParser` and `AnimeEpisodeParser` are also using the same XPath with meta tag, so I can update those too if that's what you're referring to.

I was referring to any other parser where there is any parsing of "description" or "synopsis" where the `\n` breakpoints would matter.

**Comment by ToshY** [comment]
@pushrbx 

After looking over at leftover parsers again, the only parsers that use meta tags for descriptions/synopsis/about are those 2 mentioned before. 

The more I tested those parsers, the less I wanted to change them due the following findings:

1. For manga:
   * With synopsis ([example](https://myanimelist.net/manga/2)): XPath `//span[@itemprop='description']` exists (note `span` here instead of `p`).
   * Without synopsis ([example](https://myanimelist.net/manga/149089)): XPath `//span[@itemprop='description']` does not exist.
2. For anime episode:
   * With synopsis ([example](https://myanimelist.net/anime/44511/Chainsaw_Man/episode/3)): No simple XPath like anime of manga can be used directly. Synposis starts somewhere around `//h2[contains(@class, 'fw-b')]`.
   * Without synopsis for existing episode ([example](https://myanimelist.net/anime/44511/Chainsaw_Man/episode/4)): Contains both `//h2[contains(@class, 'fw-b')]` and `//div[contains(@class, 'badresult al')]`.
   * Without synopsis for non-existing episode ([example](https://myanimelist.net/anime/44511/Chainsaw_Man/episode/12)): Contains only `//div[contains(@class, 'badresult al')]`.

Because each media type behaves differently (regarding synopsis), I think it would be more manageable to cut it into separate PRs, each one fixing it for a specific media type and updating its corresponding test. 

In order to **not** broaden the scope of the initial issue, I want to keep the PR as is.

**Comment by pushrbx** [comment]
Understood, then it's fine like this. Just one thing

File: src/Parser/Anime/AnimeParser.php [hunk 1]
```diff
@@ -90,10 +90,10 @@ public function getImageURL(): string
     public function getSynopsis(): ?string
     {
         $synopsis = JString::cleanse(
-            $this->crawler->filterXPath('//meta[@property=\'og:description\']')->attr('content')
+            $this->crawler->filterXPath('//p[@itemprop=\'description\']')->html()
```
Comment (pushrbx) [inline src/Parser/Anime/AnimeParser.php:93]: is there any chance that instead of `p` there would be a different tag?

Comment (ToshY) [reply src/Parser/Anime/AnimeParser.php:93]: For anime no. The `p` tag is there for shows that have a synopsis ([example](https://myanimelist.net/anime/28977/Gintama%C2%B0)) as well as if there is no synopsis ([example](https://myanimelist.net/anime/24857/Akagaki_Genzou__Tokuri_no_Wakare)). 

Could also change it to just `[@itemprop=\'description\']` to make sure it keeps matching even if the MAL developpers decide to change the tag to something else (like `span` at manga).

**Review by pushrbx — APPROVED** [review]

**Comment by ToshY** [comment]
@irfan-dahir is it possible to tag a new release for this?

**Pull request merged**